src/lib/y2storage/proposal/space_distribution.rb

```diff
@@ -47,7 +49,9 @@ def initialize(volumes_by_disk_space)
         end
         @spaces.freeze
         spaces_by_disk.each do |disk, spaces|
-          set_num_logical_for(spaces, disk.partition_table)
+          disk.as_not_empty do
```

Comment:
No.

First, we already have support for FreeDiskSpace in empty disks. See https://github.com/yast/yast-storage-ng/pull/152

Second, even if we wouldn't have that, the usage of `as_not_empty` ensures we treat the empty and unused disks as disks with an empty GPT table (that's what they will become in the end if we end up creating partitions there). So we will count all the space except the one used by the (future) data structures.